master/internal/api_user_intg_test.go

```diff
@@ -924,3 +926,207 @@ func getActivityEntry(ctx context.Context, userID model.UserID, entityID int32)
+		UserId: int32(userID),
+	})
+	require.NoError(t, err)
+	require.NotNil(t, token)
+
+	tokenInfo, err := api.GetUserLongLivedToken(ctx, &apiv1.GetUserLongLivedTokenRequest{
+		UserId: int32(userID),
+	})
+	require.NoError(t, err)
+	require.NotNil(t, tokenInfo)
+	require.Equal(t, int32(userID), tokenInfo.LongLivedTokenInfo.UserId)
+}
+
+func TestAuthzLongLivedToken(t *testing.T) {
+	api, authzUsers, curUser, ctx := setupUserAuthzTest(t, nil)
+
+	// POST API Auth check
+	expectedErr := status.Error(codes.PermissionDenied, "canCreateUsersOwnToken")
+	authzUsers.On("CanCreateUsersOwnToken", mock.Anything, curUser).
+		Return(fmt.Errorf("canCreateUsersOwnToken")).Once()
+
+	_, err := api.PostLongLivedToken(ctx, &apiv1.PostLongLivedTokenRequest{})
+	require.Equal(t, expectedErr.Error(), err.Error())
+
+	// GET API Auth check
+	expectedErr = status.Error(codes.PermissionDenied, "canCreateUsersOwnToken")
+	authzUsers.On("CanCreateUsersOwnToken", mock.Anything, curUser).
+		Return(fmt.Errorf("canCreateUsersOwnToken")).Once()
+
+	_, err = api.GetLongLivedToken(ctx, &apiv1.GetLongLivedTokenRequest{})
+	require.Equal(t, expectedErr.Error(), err.Error())
+}
+
+func TestAuthzUserLongLivedToken(t *testing.T) {
+	api, authzUsers, curUser, ctx := setupUserAuthzTest(t, nil)
+
+	// POST API Auth check
+	expectedErr := status.Error(codes.PermissionDenied, "canCreateUsersToken")
+	authzUsers.On("CanCreateUsersToken", mock.Anything, curUser, curUser).
+		Return(fmt.Errorf("canCreateUsersToken")).Once()
+
+	_, err := api.PostUserLongLivedToken(ctx, &apiv1.PostUserLongLivedTokenRequest{
+		UserId: int32(curUser.ID),
+	})
+	require.Equal(t, expectedErr.Error(), err.Error())
+
+	// GET API Auth check
+	expectedErr = status.Error(codes.PermissionDenied, "canCreateUsersToken")
+	authzUsers.On("CanCreateUsersToken", mock.Anything, curUser, curUser).
+		Return(fmt.Errorf("canCreateUsersToken")).Once()
+
+	_, err = api.GetUserLongLivedToken(ctx, &apiv1.GetUserLongLivedTokenRequest{
+		UserId: int32(curUser.ID),
+	})
+	require.Equal(t, expectedErr.Error(), err.Error())
+}
+
+func getLongLivedTokensEntry(ctx context.Context, userID model.UserID) (bool, error) {
+	return db.Bun().NewSelect().Table("long_lived_tokens").
+		Where("user_id = ?", userID).Exists(ctx)
+}
+
+func testSetLifespan(ctx context.Context, t *testing.T, userID model.UserID, lifespan string) error {
+	var expiresAt, createdAt time.Time
+	err := db.Bun().NewSelect().
+		Table("long_lived_tokens").
+		Column("expires_at", "created_at").
+		Where("user_id = ?", userID).
+		Scan(ctx, &expiresAt, &createdAt)
+	if err != nil {
+		return fmt.Errorf("Error getting the set lifespan, creation time")
+	}
+
+	duration, err := time.ParseDuration(lifespan)
+	if err != nil {
+		return fmt.Errorf("Invalid duration format")
+	}
+
+	durationDifference := expiresAt.Sub(createdAt)
+
+	// Define a small threshold to account for minor differences
+	threshold := 1 * time.Millisecond
```

Comment:
We might be able to avoid this if instead we are doing the created at & expiration in the insertion query; that way we could calculate the exact lifespan leveraging the difference between `created_at` and `expires_at`

Reply:
Done.